### ISSUE TITLE
update runc dependency

### ISSUE THREAD

**Issue by gfvirga** [helm/helm#7846]
Hello, any way to update the go mod github.com/opencontainers/runc to use the latest version?
It's getting listed in our vulnerability scanners for us.

Output of `helm version`:
```
 # helm version
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

Output of security scanner.
1.0.0-rc7
Summary
runc through 1.0-rc6, as used in Docker before 18.09.2 and other products, allows attackers to overwrite the host runc binary (and consequently obtain host root access) by leveraging the ability to execute a command as root within one of these types of containers: (1) a new container with an attacker-controlled image, or (2) an existing container, to which the attacker previously had write access, that can be attached with docker exec. This occurs because of file-descriptor mishandling, related to /proc/self/exe.
CVEs
CVE-2019-5736 (CVSS v2 Score : 9.3/AV:N/AC:M/Au:N/C:C/I:C/A:C) (CVSS v3 Score : 8.6/CVSS:3.0/AV:L/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H)
References
http://www.securityfocus.com/bid/106976
https://tools.cisco.com/security/center/content/CiscoSecurityAdvisory/cisco-sa-20190215-runc
https://www.synology.com/security/advisory/Synology_SA_19_06
Show All
Infected Component
github.com/opencontainers/runc
Source Version
0.1.1
Infected Versions
0.0.9 ≤ Version ≤ 1.0.0-rc6
Watch Name
HIGH-WATCH-CI
Matched Policies
HIGH-SECURITY-NOTIFY-ONLY-POLICY (Violated Rule: HIGH-SECURITY-RULE-NOTIFY)

**Comment by bacongobbler** [comment]
Hi @gfvirga. In the future, please reach out using the proper channels to disclose potential security risks. See https://github.com/helm/helm/blob/master/CONTRIBUTING.md#reporting-a-security-issue for more details.

Okay, back to the issue.

Looking at Helm's source code, Helm only uses runc's as a sub-sub-dependency of [oras](https://github.com/deislabs/oras), which in turn uses [containerd](https://github.com/containerd/containerd), which finally uses runc for process execution.

However, Helm does not use containerd (or runc) to execute container runtimes **at any point** during its execution. We are only utilizing containerd's libraries to fetch and push OCI manifests to docker registries. hence the reliance on containerd.

In other words, the attack vector you described above does not apply to Helm.

Please feel free to submit a PR to bump Helm's sub-sub-dependencies to a higher version if you wish. As it currently stands there does not appear to be any security risk with the current version.

Again, I would like to re-iterate to **please** use the proper security channels next time. It's important to us to ensure that potential security risks are not publicly disclosed until we have a patch available for general consumption.

**Comment by technosophos** [comment]
As a double-check, yes, it appears that Helm is not directly affected by the bug, which occurs only _when a container is run_, which Helm does not do at all, ever.

I have changed the issue title to better reflect the nature of the problem.

**Comment by technosophos** [comment]
@jdolitsky Do you want to upstream this with ORAS? @radu-matei warned me that simply bumping runc versions could be a bad idea

**Comment by gfvirga** [comment]
@bacongobbler, thank you responding so quickly and I apologize for not using the proper channel, it won't happen again.

**Comment by bacongobbler** [comment]
As a closing action item for this ticket, please feel free to follow up with the upstream project taking containerd as a dependency (oras). Once that's done we can update oras, and runc will be updated automatically.

As @technosophos alluded to earlier, @jdolitsky may know more about what's involved with a bump to runc.

Thanks!

**Comment by jdolitsky** [comment]
I've added a ticket on the ORAS side. Apologies I didnt see this earlier
https://github.com/deislabs/oras/issues/156
